### PR TITLE
Diffusion Imaging in Python (DIPY)

### PR DESCRIPTION
**Pull request by stephanmeesters**
This PR adds the installation for DIPY (https://github.com/nipy/dipy)

Dependencies:
- Python
- Numpy
- Nibabel
- Clang-OMP (to compile Cython with multithreading, recommended)
- VTK (with Python, recommended)
- Cython

For testing:
`brew install stephanmeesters/python/dipy`

### CODE REVIEW

**Comment by tdsmith** [comment]
Thanks for the PR! I'll review this sometime this weekend.

File: dipy.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    sha256 "6de44d8c482128efc12334641347a9c3e5098d807dd3c69e867fa8f84ec2a3f1"
+  end
+
+  def install
+    ENV.prepend_create_path "PYTHONPATH", libexec/"vendor/lib/python2.7/site-packages"
+    %w[nibabel cython].each do |r|
+      resource(r).stage do
+        system "python", *Language::Python.setup_install_args(libexec/"vendor")
+      end
+    end
```
Comment (tdsmith) [inline dipy.rb]: This should be within the `each_python` block

File: dipy.rb [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+    Language::Python.each_python(build) do |python, version|
+      ENV["PYTHONPATH"] = Formula["dipy"].opt_lib/"python#{version}/site-packages"
```
Comment (tdsmith) [inline dipy.rb]: This line is redundant with the line beneath and can be removed

**Comment by stephanmeesters** [comment]
Thanks for the review, that ```each_python``` fixed a lot of problems.

Question: should I create the bottles myself or is this done through some kind of bot? I can only build a bottle for El Capitan.

File: dipy.rb [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  url "https://github.com/nipy/dipy/archive/0.11.0.tar.gz"
+  sha256 "b9b7c19ccf9d5087f2bdea86478319a5355bb4d4c5cb23afd7d85b09048d1716"
+  head "https://github.com/nipy/dipy.git"
+
+  option "without-python", "Build without python2 support"
+
+  depends_on :python => :recommended if MacOS.version <= :snow_leopard
+  depends_on :python3 => :optional
+  depends_on "cmake" => :build
+  depends_on "clang-omp" => :recommended
```
Comment (tdsmith) [inline dipy.rb]: There's no longer a clang-omp formula, sorry.

**Comment by stephanmeesters** [comment]
I've removed the Clang-OMP part since its not essential. It should be good to go I think.

**Comment by tdsmith** [comment]
@BrewTestBot test this please

File: dipy.rb [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  depends_on :python3 => :optional
+  depends_on "cmake" => :build
+
+  numpy_options = []
+  numpy_options << "with-python3" if build.with? "python3"
+  depends_on "homebrew/python/numpy" => numpy_options
+  depends_on "homebrew/python/scipy" => numpy_options
+  depends_on "homebrew/science/vtk" => :recommended
+
+  option "without-check", "Don't run tests during installation"
```
Comment (tdsmith) [inline dipy.rb]: `without-test`, please

File: dipy.rb [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+      system python, "setup.py", "build"
+      system python, *Language::Python.setup_install_args(prefix)
+    end
+  end
+
+  test do
+    Language::Python.each_python(build) do |python, _version|
+      system python, "-c", "import dipy; assert dipy.test().wasSuccessful()"
+    end
```
Comment (tdsmith) [inline dipy.rb]: This is timing out. Let's do this in the `install` method (guarded by a check for `build.with?("test")`) and just try `import dipy` in the test.

**Comment by MikeMcQuaid** [comment]
Apologies, we're migrating this tap to homebrew/core. Please either remove the homebrew/science dependency and submit there or submit it to homebrew/science if it makes more sense with that dependency.

**Pull request closed without merging**